pkg/nodeagent/controller/operatingsystemconfig/changes.go

```diff
@@ -139,6 +151,55 @@ func computeOperatingSystemConfigChanges(log logr.Logger, fs afero.Afero, newOSC
 		changes.Files,
 	)
 
+	if oldOSC.Spec.InPlaceUpdates != nil && newOSC.Spec.InPlaceUpdates != nil {
+		if oldOSC.Spec.InPlaceUpdates.OperatingSystemVersion != newOSC.Spec.InPlaceUpdates.OperatingSystemVersion &&
+			currentOSVersion != nil && *currentOSVersion != newOSC.Spec.InPlaceUpdates.OperatingSystemVersion {
```

Comment:
```suggestion
		if currentOSVersion != nil && *currentOSVersion != newOSC.Spec.InPlaceUpdates.OperatingSystemVersion {
```
Shouldn't this check be sufficient?  

Otherwise there could be strange update behaviors like
1. `old == 1.1`, `new == 1.1`, `current == 1.0`  -> no update
2. `old == 1.1`, `new == 1.2`, `current == 1.0`  -> update

Reply:
I don't think these are possible cases, for
1. We persist the new OSC only when the new version is applied.
2. We don't allow subsequent in-place updates to the workerpool.

Nevertheless, the check you suggested should be sufficient.